src/message/attachment.rs

```diff
@@ -0,0 +1,311 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Attachment {
+    /// A JSON object that gives access to the actual content of the attachment.
+    /// Can be based on base64, json or external links.
+    pub data: AttachmentData,
+
+    /// Identifies attached content within the scope of a given message.
+    ///  Recommended on appended attachment descriptors. Possible but generally unused
+    ///  on embedded attachment descriptors. Never required if no references to the attachment
+    ///  exist; if omitted, then there is no way to refer to the attachment later in the thread,
+    ///  in error messages, and so forth. Because id is used to compose URIs, it is recommended
+    ///  that this name be brief and avoid spaces and other characters that require URI escaping.
+    pub id: Option<String>,
+
+    /// A human-readable description of the content.
+    pub description: Option<String>,
+
+    /// A hint about the name that might be used if this attachment is persisted as a file.
+    /// It is not required, and need not be unique. If this field is present and mime-type is not,
+    /// the extension on the filename may be used to infer a MIME type.
+    pub filename: Option<String>,
```

Comment:
`filename` MUST be present
https://identity.foundation/didcomm-messaging/spec/#reference-2

Reply:
https://identity.foundation/didcomm-messaging/spec/#reference-2
> filename: A hint about the name that might be used if this attachment is persisted as a file. **It is not required,** and need not be unique.

Reply:
Agree with @ashcherbakov